src/Phinx/Migration/AbstractMigration.php

```diff
@@ -62,11 +62,14 @@
     /**
      * Class Constructor.
      *
-     * @param int $version Migration Version
+     * @param int                  $version Migration Version
+     * @param OutputInterface|null $output
      */
-    final public function __construct($version)
+    final public function __construct($version, OutputInterface $output = null)
     {
         $this->version = $version;
+        $this->output = $output;
```

Comment:
In order to not make this a breaking change, this needs to call `$this->setOutput($output)`.